### PR TITLE
Vcpkg fixes

### PR DESCRIPTION
**Pull request by antkmsft**
With this fix, vcpkg CI looks good - https://github.com/microsoft/vcpkg/pull/17119

### CODE REVIEW

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/vcpkg/vcpkg.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   "license": "MIT",
   "dependencies": [
     {
-      "name": "azure-storage-common-cpp",
+      "name": "azure-storage-blobs-cpp",
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-datalake/vcpkg/vcpkg.json:15]: What motivated this change @antkmsft ?

@Jinming-Hu is this dependency hierarchy correct?

File: sdk/core/azure-core/src/io/body_stream.cpp [hunk 2]
```diff
@@ -128,7 +128,10 @@ FileBodyStream::FileBodyStream(const std::string& filename)
   }
   catch (std::exception&)
   {
-    CloseHandle(fileHandle);
+    if (fileHandle != INVALID_HANDLE_VALUE)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/io/body_stream.cpp:131]: Do we really need this check or can we close a handle even if it is `INVALID_HANDLE_VALUE`?

**Review by ahsonkhan — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/vcpkg/vcpkg.json (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/vcpkg/vcpkg.json:15]: Yes

File: sdk/core/azure-core/src/io/body_stream.cpp (hunk 2 above)
Comment (antkmsft) [reply sdk/core/azure-core/src/io/body_stream.cpp:131]: For the limited amount of time I researched, I haven't found a definitive answer, not closing it is safer - https://social.msdn.microsoft.com/Forums/en-US/ede8051a-ec1e-44b2-a42a-11332421d23e/is-quotclosehandle0quot-always-safe?forum=windowsgeneraldevelopmentissues

**Pull request merged**